lib/testscript_engine/testscript_runnable.rb

```diff
@@ -194,7 +194,16 @@ def get_fixture_from_ref(reference)
 
     ref = reference.reference
     return warning(:no_reference) unless ref
-    return warning(:unsupported_ref, ref) if ref.start_with? 'http'
+
+    if ref.start_with? 'http'
```

Comment:
I'm surprised you didn't use the FHIR client here. I would expect that fixtures will be FHIR resources and so the servers hosting the remote resources will be FHIR servers and should be accessed as such. Was there a reason you went with HTTP here?

Reply:
A simple reason was the engine should support non FHIR fixture. I wonder FHIR Client support non-FHIR resources as well. Do you have an idea?

Non FHIR fixture is currently supported in local only. I wonder if we have good test cases for non-FHIR fixture so that I can test it in remote as well. Thanks for reminding this actually, I will go back to it and add the test as well.

It is also considerable to separate the method, depending on non-FHIR fixture and FHIR fixture, use FHIR Client and general http client for the other. Do you think it's better?

Reply:
I took a look at the local non-FHIR fixture part and it's doesn't look good. It figures out non-FHIR fixtures if

!FHIR.from_contents(contents)

I'm inclined to handle only FHIR fixture remotely. If it is local, at least user can mark it using folder name (/fixture_nonfhir). If it's remote, it's very difficult to mark it and also there are lots of potential risk of importing non-FHIR fixtures (in other words anything from the web) and use it in testing. What do you think?

Also the recent commit changed to use FHIR Client.

Reply:
I think it would be fine to assume that remote fixtures are FHIR instances.

On the other hand, I think the right way to support non-FHIR fixtures is to use an extension to make the intention on the part of the test author explicit:
```
"fixture": [
    {
      "id": "fixture-patient-create",
      "autocreate": false,
      "autodelete": false,
      "extension": [
          {
            "url": "urn:mitre:non-fhir-fixture",
            "valueBoolean": true
          }
       ],
      "resource": {
        "reference": "http://hl7.org/fhir/patient-example.json",
        "display": "Peter Chalmers"
      }
    }
  ]
```

That said, I don't think this stage of the project is the right time to start introducing extensions, since we'd really want an IG to document them like Touchstone has.  So I think probably continuing as you have and disallowing remote non-FHIR fixtures is the right path for the time being.

Reply:
> I think it would be fine to assume that remote fixtures are FHIR instances.
> 
> On the other hand, I think the right way to support non-FHIR fixtures is to use an extension to make the intention on the part of the test author explicit:
> 
> ```
> "fixture": [
>     {
>       "id": "fixture-patient-create",
>       "autocreate": false,
>       "autodelete": false,
>       "extension": [
>           {
>             "url": "urn:mitre:non-fhir-fixture",
>             "valueBoolean": true
>           }
>        ],
>       "resource": {
>         "reference": "http://hl7.org/fhir/patient-example.json",
>         "display": "Peter Chalmers"
>       }
>     }
>   ]
> ```
> 
> That said, I don't think this stage of the project is the right time to start introducing extensions, since we'd really want an IG to document them like Touchstone has. So I think probably continuing as you have and disallowing remote non-FHIR fixtures is the right path for the time being.

Yes, I agree with you that extension inside a fixture would be plausible way to do non-fhir fixture. Also resonate this isn't right time to do it. I will go ahead and finish.